Server/ReasnAPI/ReasnAPI/Services/InterestService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using ReasnAPI.Exceptions;
+using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
+using ReasnAPI.Mappers;
+
+namespace ReasnAPI.Services;
+
+public class InterestService(ReasnContext context)
+{
+    public InterestDto CreateInterest(InterestDto interestDto)
+    {
+        var interest = context.Interests.FirstOrDefault(r => r.Name == interestDto.Name);
+        if (interest is not null)
+        {
+            throw new BadRequestException("Interest already exists");
+        }
+
+        var newInterest = interestDto.ToEntity();
+
+        context.Interests.Add(newInterest);
+        context.SaveChanges();
+        return interestDto;
+    }
+
+    public InterestDto UpdateInterest(int interestId, InterestDto interestDto)
+    {
+        var interest = context.Interests.FirstOrDefault(r => r.Id == interestId);
+        if (interest is null)
+        {
+            throw new NotFoundException("Interest not found");
+        }
+
+        interest.Name = interestDto.Name;
+
+        context.Interests.Update(interest);
+        context.SaveChanges();
+        return interestDto;
+    }
+
+    public void DeleteInterest(int id)
+    {
+        var interest = context.Interests.FirstOrDefault(r => r.Id == id);
+        if (interest is null)
+        {
+            throw new NotFoundException("Interest not found");
+        }
+
+        var userInterest = context.UserInterests.FirstOrDefault(r => r.InterestId == id);
+        if (userInterest is not null)
+        {
+            throw new BadRequestException("Interest is in use");

```

Comment:
Assuming this is only for user, shouldn't we delete all userinterest and then interest? I think as an admin it would be greate to have control of all intrests and no need to go on DB to edit it manually or one by one per user

Reply:
created separate method to force delete intrest